conf4j-core/src/test/java/com/sabre/oss/conf4j/factory/ConventionConfigurationFactoryTest.java

```diff
@@ -79,6 +79,7 @@ private void validate() {
         assertThat(configuration.getTimeout().getConnectTimeout()).isEqualTo(1000);
         assertThat(configuration.getTimeout().getReadTimeout()).isEqualTo(2000);
         assertThat(configuration.getOtherTimeouts().size()).isEqualTo(3);
+        assertThat(configuration.getUserNameWithUrl()).isEqualTo("user " + "http://some.url");
```

Comment:
General note for the tests. This particular test is designed to test the functionality of _ConventionConfigurationFactory_. Tests related to _default_ method should be defined elsewhere. 

I would suggest _AbstractConfigurationFactoryTest_ 

Here is an example test method and associated configuration:

```java
 @Test
    public void shouldSupportDefaultMethods() {
        when(source.getValue(anyString(), any())).thenReturn(absent());
        when(source.getValue("configuration.firstName", null)).thenReturn(present("John"));
        when(source.getValue("configuration.lastName", null)).thenReturn(present("Doe"));

        // when
        ConfigurationWithDefaultMethods configuration = factory.createConfiguration(ConfigurationWithDefaultMethods.class, source);

        // then
        assertThat(configuration.getFirstName()).isEqualTo("John");
        assertThat(configuration.getLastName()).isEqualTo("Doe");
        assertThat(configuration.getFullName()).isEqualTo("John Doe");
    }

    @Key("configuration")
    public interface ConfigurationWithDefaultMethods {
        String getFirstName();

        String getLastName();

        default String getFullName() {
            return getFirstName() + ' ' + getLastName();
        }
    }
```